### ISSUE TITLE
Rethink progress bar UI.

### ISSUE THREAD

**Issue by lgarron** [cubing/alg.cubing.net#20]
Something like this YouTube-style slider would be awesome, since people instantly recognize it: http://codepen.io/jonomcleish/pen/rDIsJ

However, I want the CSS to be simple, with a proper `range` fallback for browsers that don't support fancy styling.

Also, mobile would benefit from a very large target area, and being able to click anywhere on the bar to jump there (right now, you have to hit the thumb target before moving an input range on iOS).

To summarize:
- Easy to recognize as a slider that you can click/touch and drag.
- Simple implementation with an obvious fallback (i.e. range slider).
- Practical to use on mobile.
